src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  let copyState = { ...state };
+
+  actions.forEach(action => {
+    if (action.type === 'addProperties') {
```

Comment:
Do it with `switch/case`
```suggestion
   //
```